### DISCUSSION TITLE
Does Pipeline.from_pretrained download the model each time or uses a cache?

### DISCUSSION THREAD

**Discussion by lsshawn** [lhotse-speech/lhotse#984] [Q&A]
When we run `Pipeline.from_pretrained` using hugging face, does it download the model every time? Or will it read from my cache?

I'm trying to make this code run faster:

```
pipeline = Pipeline.from_pretrained(
        "pyannote/segmentation",
        use_auth_token="TOKEN",
    )

for wav in wavFiles:
    pipeline(wav)
```

**Comment by pzelasko** [comment]
Hi, I think this question is more appropriate for either pyannote or huggingface communities.
